thorlcr/activities/thdiskbaseslave.cpp

```diff
@@ -551,17 +551,17 @@ void CDiskWriteSlaveActivityBase::abort()
 
 void CDiskWriteSlaveActivityBase::serializeStats(MemoryBuffer &mb)
 {
-    CRuntimeStatisticCollection activeStats(diskWriteRemoteStatistics);
-    {
-        CriticalBlock block(outputCs);
-        activeStats.set(closedPartFileStats);
-        mergeStats(activeStats, outputIO, diskWriteRemoteStatistics);
-    }
-    stats.set(activeStats); // replace disk write stats
     stats.setStatistic(StPerReplicated, replicateDone);
     PARENT::serializeStats(mb);
 }
 
+// NB: always called within statsCs crit
+void CDiskWriteSlaveActivityBase::gatherActiveStats(CRuntimeStatisticCollection &activeStats) const
+{
+    mergeStats(activeStats, outputIO, diskWriteRemoteStatistics);
```

Comment:
can outputIO change while this is being called (and need protecting with outputCs?

Reply:
changes to outputIO are being protected by the 'statsCS' crit in the base activity class, which is also held whilst this method is called.